### PR TITLE
Login01

### PR DESCRIPTION
**Pull request by FloresFelix**
maquetado del login

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                    editTextEmail.setError("Debe ingrese el correo")
+                } else if (editTextPassword.text.toString().isEmpty()) {
+                    editTextPassword.setError("Debe ingresar la contraseña")
+                } else {
+                    if (!android.util.Patterns.EMAIL_ADDRESS.matcher(editTextEmail.text.toString()).matches()) {
+                        editTextEmail.setError("El correo debe tener el formato: example@domain.com")
+                    } else {
+                        presenter.onClickLogin(editTextEmail.text.toString(), editTextPassword.text.toString())
+                    }
+                }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt]: This belongs to the card named "Validaciones" 😅 

Also, you should do these validations in the presenter. Remember to never put logic inside the fragment.

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package ar.com.wolox.android.example.ui.login
+
+import androidx.core.widget.addTextChangedListener
+import ar.com.wolox.android.R
+import ar.com.wolox.android.databinding.FragmentLoginBinding
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+
+class LoginFragment private constructor() : WolmoFragment<FragmentLoginBinding, LoginPresenter>() {
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt]: Remember you have to implement the LoginView. 
You can implement classes in kotlin by adding after a comma:
`class LoginFragment : WolmoFragment<FragmentLoginBinding, LoginPresenter>(), LoginView`

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginView.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    fun dummyFunction() {}
+
+    fun dummyFunction2() {}
+
+    fun dummyFunction3() {}
+
+    fun dummyFunction4() {}
+
+    fun dummyFunction5() {}
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginView.kt]: Remember to delete these lines after doing the navigation card

File: app/src/main/res/layout/fragment_example.xml [hunk 4]
```diff
@@ -51,6 +51,7 @@
 			android:maxLines="1"
 			app:layout_constraintBottom_toTopOf="@id/loginButton"
 			app:layout_constraintEnd_toEndOf="parent"
+			app:layout_constraintHorizontal_bias="1.0"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_example.xml:54]: You don't need this change

File: app/src/main/res/layout/fragment_login.xml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".example.ui.login.LoginFragment">
+
+
+        <ImageView
+            android:id="@+id/imageViewLogo"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="32dp"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: Don't use hardcoded dimens. You can add this dimen inside the `dimens.xml` file to reuse it.

File: app/src/main/res/layout/fragment_login.xml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <EditText
+            android:id="@+id/editTextEmail"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="92dp"
+            android:layout_marginEnd="24dp"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: Solve hardcoded dimens

File: app/src/main/res/layout/fragment_login.xml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            android:inputType="text"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/imageViewLogo" />
+
+        <EditText
+            android:id="@+id/editTextPassword"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="24dp"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: Hardcoded dimen

File: app/src/main/res/layout/fragment_login.xml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            android:inputType="textPassword"
+            app:layout_constraintEnd_toEndOf="@+id/editTextEmail"
+            app:layout_constraintStart_toStartOf="@+id/editTextEmail"
+            app:layout_constraintTop_toBottomOf="@+id/editTextEmail" />
+
+        <Button
+            android:id="@+id/buttonLogin"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="32dp"
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_login.xml]: Check these hardcoded dimens in the rest of the file

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -13,8 +13,17 @@
...
     <string name="fragment_example_login">Login</string>
-	<string name="fragment_request_search">Search</string>
+    <string name="fragment_request_search">Search</string>
     <string name="fragment_request_post_id">Post id</string>
     <string name="fragment_request_invalid_input">Invalid input, please enter a valid id.</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
+    <string name="first_name">First Name (recomended)</string>
+    <string name="last_name">Last Name (recomended)</string>
+    <string name="sing_up">SING UP</string>
```
Comment (ffasitella) [inline app/src/main/res/values/strings.xml]: Typo, it's `SIGN UP`

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -13,8 +13,17 @@
...
+    <string name="log_in">LOG IN</string>
+    <string name="email">Email</string>
+    <string name="password">contraseña</string>
+    <string name="terms"><![CDATA[By joining you agree to our Terms & Conditions]]></string>
```
Comment (ffasitella) [inline app/src/main/res/values/strings.xml]: You need to make the "Terms & Conditions" part underlined

**Review by ffasitella — CHANGES_REQUESTED** [review]
I saw you didn't create an activity. Try creating a `LoginActivity` and setting it as the main activity.
Also, we usually try to write useful information in the Pull Request's description. Something like:
```
### Summary
- List of features you created in this PR

### Screenshot
- A screenshot or video of your screen

### Trello card
- Link to your trello card
```
I left some comments! Feel free to ask us anything.

**Comment by FloresFelix** [comment]
genail! Voy Revisar!

File: app/build.gradle [hunk 11]
```diff
@@ -152,4 +151,7 @@ dependencies {
     // Android KTX
     implementation "androidx.core:core-ktx:$android_ktx_version"
     implementation "androidx.fragment:fragment-ktx:$android_ktx_version"
+
```
Comment (jcruz-wolox) [inline app/build.gradle]: Remove this empty lines

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    override fun setListeners() {
+        with(binding) {
+
+            editTextEmail.addTextChangedListener {}
+            editTextPassword.addTextChangedListener {}
+            buttonLogin.setOnClickListener {
+
+                // Validacion
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt]: Comments in English

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+                }
+            }
+            buttonSignup.setOnClickListener { presenter.onClickSingup() }
+            textViewTerms.setOnClickListener { presenter.onClickTerms() }
+        }
+    }
+
+    companion object {
+
+    fun newInstance() = LoginFragment()
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt]: Indent this line

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+                    }
+                }
+            }
+            buttonSignup.setOnClickListener { presenter.onClickSingup() }
+            textViewTerms.setOnClickListener { presenter.onClickTerms() }
+        }
+    }
+
+    companion object {
+
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.kt:31]: Remove empty line

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+
+class LoginPresenter constructor() : BasePresenter<LoginView>() {
+
+    fun onClickLogin(fistName: String, lastName: String) {
+        view?.dummyFunction()
+    }
+
+    fun onClickSingup() {
+        view?.dummyFunction2()
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt]: Don't use numbers like part of names, you should use literals, so, if you going to use dummyFunction2 should rename to dummyFunctionTwo

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    fun onClickTerms() = view?.gotToWeb()
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt]: you have a typo in "gotToWeb()" correct with goToWeb()

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginView.kt [hunk 17]
```diff
@@ -0,0 +1,16 @@
+package ar.com.wolox.android.example.ui.login
+
+interface LoginView {
+
+    fun dummyFunction() {}
+
+    fun dummyFunction2() {}
```
Comment (jcruz-wolox) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginView.kt]: The same that above with numbers in names

File: app/src/main/res/layout/fragment_login.xml [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+
+<layout xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".example.ui.login.LoginFragment">
+
```
Comment (jcruz-wolox) [inline app/src/main/res/layout/fragment_login.xml:9]: delete empty line

**Review by jcruz-wolox — CHANGES_REQUESTED** [review]
I agree with the Fede's comments.
good job, you're doing fine!

**Comment by FloresFelix** [comment]
Quiero que saber mi pueden ayudar en esto... Tengo lleno la bandeja de
entrada de outlook con mensajes que tengo que ir a *myTimeandExpenses and
submit your time as soon possible.* Me pueden dar una breve explicación de
como cargar esos datos, sobre todo en en la pestaña PREFERENCES que es
donde me pide que ponga datos si o si, Gracias

**Review by ffasitella — APPROVED** [review]
Good job! 👍🏽

**Review by jcruz-wolox — APPROVED** [review]
Nice!

**Pull request merged**